src/plugins/workspace/public/workspace_client.ts

```diff
@@ -117,13 +117,21 @@ export class WorkspaceClient implements IWorkspaceClient {
         perPage: 999,
         permissionModes: [WorkspacePermissionMode.LibraryWrite],
       });
-      if (resultWithWritePermission?.success) {
+      const resultWithOwnerPermission = await this.list({
+        perPage: 999,
+        permissionModes: [WorkspacePermissionMode.Write],
+      });
```

Comment:
Can we use Promise.all to make the calls in parallel?

Reply:
updated